### PR TITLE
Fix additional clang-tidy 16 errors, remove some obsolete suppressions

### PR DESCRIPTION
**Pull request by Robyt3**
Fix additional clang-tidy errors that are found when running the current checks with clang-tidy 16.

Remove some obsolete suppressions and fix the errors instead.

## Checklist

- [ ] Tested the change ingame
- [ ] Provided screenshots if it is a visual change
- [ ] Tested in combination with possibly related configuration options
- [ ] Written a unit test (especially base/) or added coverage to integration test
- [ ] Considered possible null pointers and out of bounds array indexing
- [ ] Changed no physics that affect existing maps
- [ ] Tested the change with [ASan+UBSan or valgrind's memcheck](https://github.com/ddnet/ddnet/#using-addresssanitizer--undefinedbehavioursanitizer-or-valgrinds-memcheck) (optional)

### CODE REVIEW

File: src/engine/client/client.cpp [hunk 1]
```diff
@@ -4564,23 +4564,23 @@ int main(int argc, const char **argv)
 
 	const bool RandInitFailed = secure_random_init() != 0;
 	if(!RandInitFailed)
-		CleanerFunctions.push([]() { secure_random_uninit(); });
+		CleanerFunctions.emplace([]() { secure_random_uninit(); });
```
Comment (heinrich5991) [inline src/engine/client/client.cpp:4567]: Does this produce code bloat by template-instantiating a new `emplace` function for every lambda?

**Review by def- — APPROVED** [review]
Can we switch the CI to run clang-tidy 16 too so the state stays green?

File: src/engine/client/client.cpp (hunk 1 above)
Comment (Robyt3) [reply src/engine/client/client.cpp:4567]: At least in a minimal example, using `emplace` results in significantly less assembly with gcc and clang: https://godbolt.org/z/3be4966n7

**Comment by Robyt3** [comment]
> Can we switch the CI to run clang-tidy 16 too so the state stays green?

It seem the `clang-tidy-16` package is only available starting with Ubuntu 23.04, while the GitHub runners are on version 22.04.

https://packages.ubuntu.com/search?keywords=clang-tidy

**Pull request merged**